### PR TITLE
tests: Reduce usage of other.do_smart_test. NFC

### PR DESCRIPTION
**Pull request by sbc100**
This change removes all the unnessary usage of this method.  It's only
need when regex's are used, otherwise it functions just like the
existing `do_runf`.

I'm hoping to remove this method completely as a followup since.  It now
only has two callsites.

I've also updated to_smart_test to honor the test-specific emcc
arguments (`self.emcc_args()`) for consistency with `do_runf` and all
its derivitives.

### CODE REVIEW

File: tests/common.py [hunk 1]
```diff
@@ -1111,7 +1111,7 @@ def _build_and_run(self, filename, expected_output, args=[], output_nicerizer=No
               self.assertContained(o, js_output)
           else:
             self.assertContained(expected_output, js_output)
-            if check_for_error:
+            if assert_returncode == 0 and check_for_error:
```
Comment (kripken) [inline tests/common.py:1114]: Is this related?

**Review by kripken — COMMENTED** [review]
lgtm % question

File: tests/common.py (hunk 1 above)
Comment (sbc100) [reply tests/common.py:1114]: Yes.. because some of the tests being converted are ones that test sanitizers and therefore have "ERROR" in their expected output (but they also set assert_returncode=NON_ZERO).
